### PR TITLE
[IMP] Payment: Payment methods improvements 

### PR DESCRIPTION
**Pull request by tbe-odoo**
Description of the issue/feature this PR addresses:
- Payment methods are now verified.
- Ability to change the payment method assigned to a subscription (only the verified one).
- Generic templates added for the payment tokens (handle payments, deletions and a lot of others cases).
- Tried to do something that it's not ugly, but sorry I don't have this magic power 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment/models/payment_acquirer.py [hunk 1]
```diff
@@ -726,9 +738,93 @@ def create(self, values):
...
+            tx.s2s_do_refund()
+        return tx
 
     @api.multi
     @api.depends('name')
     def _compute_short_name(self):
         for token in self:
             token.short_name = token.name.replace('XXXXXXXXXXXX', '***')
+
+    @api.multi
```
Comment (tbe-odoo) [inline addons/payment/models/payment_acquirer.py]: I was thinking about doing something more generic and easier to update.

```python
    @api.multi
    def _get_linked_records(self):
        """ Returns the list of records currently using the records in self as payment token. """
        res = {}
        model_list = [
            # (model in which you want to search, field containing the id of the payment token)
            ('sale.subscription', 'payment_token_id'),
        ]

        for r in self:
            res[r.id] = []

            for m in model_list:
                records = self.env[m[0]].search([(m[1], '=', r.id)])
                if records:
                    res[r.id].append((records._description, records))

        return res
```

File: addons/payment/models/payment_acquirer.py [hunk 2]
```diff
@@ -726,9 +738,93 @@ def create(self, values):
                 fields_wl = set(self._fields) & set(values)
                 values = {field: values[field] for field in fields_wl}
         return super(PaymentToken, self).create(values)
+    """
+        @TBE: stolen shamelessly from there https://www.paypal.com/us/selfhelp/article/why-is-there-a-$1.95-charge-on-my-card-statement-faq554
+        Most of them are ~1.50€s
+        TODO: See this with @AL & @DBO
+    """
+    validation_amounts = {
```
Comment (dbo-odoo) [inline addons/payment/models/payment_acquirer.py]: VALIDATION_AMOUNTS

File: addons/payment/models/payment_acquirer.py [hunk 3]
```diff
@@ -726,9 +738,93 @@ def create(self, values):
...
+        currency = self.partner_id.currency_id
+
+        if self.validation_amounts.get(currency.name):
+            amount = self.validation_amounts.get(currency.name)
+        else:
+            # If we don't find the user's currency, then we set the currency to EUR and the amount to 1€50.
+            currency = self.env['res.currency'].search([('name', '=', 'EUR')])
+            amount = 1.5
+
+        if len(currency) != 1:
```
Comment (dbo-odoo) [inline addons/payment/models/payment_acquirer.py]: currency without len

File: addons/payment/models/payment_acquirer.py (hunk 1 above)
Comment (tbe-odoo) [inline addons/payment/models/payment_acquirer.py]: read group

File: addons/payment_ogone/controllers/main.py [hunk 4]
```diff
@@ -28,6 +28,16 @@ def ogone_form_feedback(self, **post):
         request.env['payment.transaction'].sudo().form_feedback(post, 'ogone')
         return werkzeug.utils.redirect(post.pop('return_url', '/'))
 
+    @http.route([
+        '/validation/ogone/accept',
```
Comment (tbe-odoo) [inline addons/payment_ogone/controllers/main.py]: payment/ogone/validate/XXXX

Comment (tbe-odoo) [inline addons/payment_ogone/controllers/main.py]: mettre ce code en dessous de s2s_create

**Pull request merged**